### ISSUE TITLE
When you set a WATCHDIR in s3sync it still tries to sync from S3 to /sync on startup

### ISSUE THREAD

**Issue by ryanwalls** [vladgh/docker_base_images#7]
Your main:

```
main(){
  if [[ ! "$S3PATH" =~ s3:// ]]; then
    log 'No S3PATH specified' >&2; exit 1
  fi

  # Run initial sync
  sync_files "$S3PATH" "$DESTINATION"

  # Exit if argument is 'once'
  if [[ "${*:-}" == 'once' ]]; then exit; fi

  # Setup inotify or cron job
  if [[ -n "$WATCHDIR" ]]; then
    watch_directory
  elif [[ -n "$CRON_TIME" ]]; then
    run_cron
  fi
}
```

Note that the initial sync is run even if WATCHDIR is specified.  I would assume that the behavior would be that if a WATCHDIR is specified, only that directory is synced to S3.  I'm assuming there should be no sync from S3 -> destination in this case.

**Comment by vladgh** [comment]
Hey @ryanwalls, that makes perfect sense.
However, this is the intended behaviour. This script is actually downloading the files first __FROM__ (in both modes: cron or inotify), and only after this initial step, it will start uploading files __TO__ S3 (periodically or when files change in the 'watchdir').
Probably this is not very clear in the way I documented the code, but I'm willing to change it, to maybe accommodate both cases. One in which it just syncs to S3, and one in which it syncs from S3 first and then syncs to S3.

**Comment by vladgh** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [https://github.com/vladgh/docker_base_images/issues/7](https://trello.com/c/dSZoSGTE/1166-https-github-com-vladgh-docker-base-images-issues-7)

**Comment by ryanwalls** [comment]
@vladgh  Ah that makes sense.  Yeah, it would be nice to have the enhancement of syncing only from host to s3 without the initial download.

**Comment by vladgh** [comment]
Instead of adding another environment variable, what about a simple check if `WATCHDIR` is empty or not? If it is empty do the initial download, if it has files in it then skip it.

**Comment by ryanwalls** [comment]
In my case I want the WATCHDIR to control the sync in all cases.   So I would want the ability to disable the initial sync whether WATCHDIR is empty or not.

**Comment by ryanwalls** [comment]
One minor point about the inotify setup: seems we have 3 locations specified S3PATH, WATCHDIR, and DESTINATION, when we are really syncing between just 2 locations.  Not sure best way to fix that or if that is intentional.  Perhaps just use WATCHDIR as the destination if it is set?

**Comment by vladgh** [comment]
That `DESTINATION` var is redundant anyway, I will get rid of it.

**Comment by ryanwalls** [comment]
Just realized we actually don't need to have the WATCHDIR control the sync.  Have another way to solve the problem.  So I would put this as a lower priority. :)  

Thanks for the tool.  Looked at a lot of s3 syncs and this was the best setup.

**Comment by vladgh** [comment]
You're welcome. If you see any improvements please feel free to open a PR.
This discussion triggered some changes though, that I will apply later today, and they might not be backward compatible.

**Comment by vladgh** [comment]
Is this working better for you?

**Comment by ryanwalls** [comment]
@vladgh So far so good.  Thanks!
